src/plugins/profiling/server/routes/search_topn.ts

```diff
@@ -76,23 +77,25 @@ export function queryTopNCommon(
       path: pathName,
       validate: {
         query: schema.object({
-          index: schema.maybe(schema.string()),
-          projectID: schema.maybe(schema.string()),
-          timeFrom: schema.maybe(schema.string()),
-          timeTo: schema.maybe(schema.string()),
+          index: schema.string(),
+          projectID: schema.string(),
+          timeFrom: schema.string(),
+          timeTo: schema.string(),
+          n: schema.maybe(schema.number()),
         }),
       },
     },
     async (context, request, response) => {
-      const { index, projectID, timeFrom, timeTo } = request.query;
+      const { index, projectID, timeFrom, timeTo, n } = request.query;
 
       try {
         return await topNElasticSearchQuery(
           context,
-          index!,
-          projectID!,
-          timeFrom!,
-          timeTo!,
+          index,
+          projectID,
+          timeFrom,
+          timeTo,
+          n ?? 100,
```

Comment:
Why not catch this in getFetchQuery(). Passing around an undefined value seems is not ideal.

Reply:
The way this is desgned is tied to the `validate` method of the KibanaRequest class, where the `Query` type is expressed.
Since the `n` query param is now the only optional one, we have to default in the backend too.
I did this because I am not entirely sure we need that _all the times_, but if we want to make it non-optional and force the default in the UI, that's fine by me.

@jbcrail thoughts?

Reply:
I think using a required query param in the Kibana backend and a hard-coded default value in the UI would be a good balance.

This is the approach used in the current prodfiler UI (the `topN` query param is hard-coded to 50 in that code base). This would simplify the Kibana backend and remove the need for handling undefined values.

We can leave `n` in `getFetchQuery` hard-coded to the default 100 for now since we may not need to make this available for configuration in the UI. It doesn't look like prodfiler UI does either.

This means the following two methods would revert to their original declarations (at least until we need to configure `n`  in the UI):

* `fetchTopN: (type: string, seconds: string)`
* `getFetchQuery(seconds: string)`